packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -851,6 +865,22 @@ abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes>
 			index,
 		);
 	}
+
+	public *values(): Generator<TreeNodeFromImplicitAllowedTypes<T>> {
```

Comment:
nit:
```suggestion
	public *values(): IterableIterator<TreeNodeFromImplicitAllowedTypes<T>> {
```

The distinction is admittedly subtle and I don't actually know in practice how it would matter, but looking at the two interfaces, `Generator` promises slightly more than than `IterableIterator`. Since JS Array.values guarantees only `IterableIterator`, we may as well be that tiny bit more conservative and do the same?

Reply:
Updated to return `IterableIterator` type instead of `Generator`